sdk/search/azure-search/src/main/java/com/azure/search/models/TextExtractionAlgorithm.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * Defines values for TextExtractionAlgorithm.
+ */
+public enum TextExtractionAlgorithm {
```

Comment:
As mentioned in the OCRSkill this type is deprecated

Reply:
since it is documented in the swagger spec, it is generated automaticly. Can be marked as deprected in the autorest config. @brjohnstmsft ?

Reply:
@navalev If the Azure API Guidelines for Java say this is the way to go, we should do it.

Reply:
@brjohnstmsft I don't see anything specific in the guidelines. I suggest removing it from the swagger spec if it is not used.

Reply:
@navalev We can't do that, because these types represent persistent resources that customers may still have. If we don't include the type, skillsets that include it will fail to deserialize. We can only remove such deprecated types when we can confirm that no instances still exist in our system.